File: important-questions/python/1.LCSOf1s.py
x = [1,1,1,0,1,1,0,1,1]

y=[0]

for i in range(1,len(x)):
  if x[i] == 0:
    y.append(i)

y.append(len(x)-1)

max=0
diff=0
current=0

for i in range(0,len(y)-2):
  diff = y[i+2] - y[i]
  if diff > max:
    max= diff
    current = y[i+1]

if len(y)==2 and x[-1]==0:
  current = len(x) -1

print(current)

File: important-questions/java/1.LCSOf1s.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        int[] arr = { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1 };

        ArrayList<Integer> zeroes = new ArrayList<>();

        zeroes.add(0);

        for (int i = 1; i < arr.length - 1; i++) {
            if (arr[i] == 0) {
                zeroes.add(i);
            }
        }

        zeroes.add(arr.length - 1);

        int[] x = zeroes.stream().mapToInt(i -> i).toArray();

        int max = 0;
        int diff = 0;
        int current = 0;

        for (int i = 0; i < x.length - 2; i++) {
            diff = x[i + 2] - x[i];
            if (diff > max) {
                max = diff;
                current = x[i + 1];
            }
        }

        if (x.length == 2 && arr[arr.length - 1] == 0) {
            current = arr.length - 1;
        }

        System.out.println(current);

    }
}